### PR TITLE
includes projections for queryRules

### PR DESCRIPTION
**Pull request by russdj**

### CODE REVIEW

File: api/source/service/mysql/STIGService.js [hunk 1]
```diff
@@ -293,26 +293,21 @@ Generalized queries for a single Rule, optionally with Check and Fix
...
-    'r.responsibility'
+    'g.groupId',
+    'g.Title as "groupTitle"'
   ]
 
   let joins = [
-    'rule r'
-  ]
-
+    'revision rev ',
```
Comment (csmig) [inline api/source/service/mysql/STIGService.js]: Please rework these joins as discussed... no need for revision table

File: api/source/service/mysql/STIGService.js [hunk 2]
```diff
@@ -324,6 +319,22 @@ exports.queryRules = async function ( ruleId, inProjection ) {
   
   // PROJECTIONS
   // Include extra columns for Rules with details OR individual Rule
+  if ( inProjection && inProjection.includes('detail') ) {
+    columns.push(
```
Comment (csmig) [inline api/source/service/mysql/STIGService.js]: Please define a groupBy array that you can push these columns to. Follow the pattern used in `queryBenchmarkRules()` [as done here](https://github.com/NUWCDIVNPT/stig-manager/blob/dac7ee85187dbf5cfd7ba3af7b2f33ac074a854c/api/source/service/mysql/STIGService.js#L146)

File: api/source/service/mysql/STIGService.js [hunk 3]
```diff
@@ -358,12 +369,6 @@ exports.queryRules = async function ( ruleId, inProjection ) {
...
 
-  if ( inProjection && inProjection.includes('detail') ) {
-    columns.push(
-      'r.iacontrols'
-    )
-  }
-
 
   // CONSTRUCT MAIN QUERY
   let sql = 'SELECT '
```
Comment (csmig) [inline api/source/service/mysql/STIGService.js:387]: Please include the groupBy array joined ",\n" here

**Review by csmig — COMMENTED** [review]
Also, please remove the rows.toJSON() method, not something we need.

File: api/source/service/mysql/STIGService.js (hunk 1 above)
Comment (russdj) [reply api/source/service/mysql/STIGService.js]: reworked order of the tables in the joins

File: api/source/service/mysql/STIGService.js (hunk 2 above)
Comment (russdj) [reply api/source/service/mysql/STIGService.js]: Added

File: api/source/service/mysql/STIGService.js (hunk 3 above)
Comment (russdj) [reply api/source/service/mysql/STIGService.js:387]: removed .toJSON() and added groupBy

**Pull request merged**